lib/temporalio/worker.rb

```diff
@@ -17,11 +17,26 @@ class Worker
     # finished) and will raise if any of the workers raises a fatal error.
     #
     # @param workers [Array<Temporalio::Worker>] A list of the workers to be run.
+    # @param stop_on_signal [Array<String>] A list of process signals for the worker to stop on.
+    #   This argument can not be used with a custom block.
     #
     # @yield Optionally you can provide a block by the end of which all the workers will be shut
     #   down. Any errors raised from this block will be re-raised by this method.
-    def self.run(*workers, &block)
-      # TODO: Add signal handling
+    def self.run(*workers, stop_on_signal: [], &block)
+      unless stop_on_signal.empty?
+        if block
+          raise ArgumentError, 'Temporalio::Worker.run accepts stop_on_signal: or a block, but not both'
```

Comment:
This colon in the error string is a bit weird, is it there intentionally?

Reply:
Yep, to indicate that it's a keyword argument, but I'm not super attached to it